Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1985,12 +2027,78 @@ public void getShortUrl(int type, Collection<String> tags, String channel, Strin
         generateShortLink(null, LINK_TYPE_UNLIMITED_USE, duration, tags, channel, feature, stage, stringifyParams(filterOutBadCharacters(params)), callback, true);
     }
 
     /**
+     * <p>Configures and requests a short URL to be generated by the Branch servers.</p>
+     *
+     * @param alias    Link 'alias' can be used to label the endpoint on the link.
+     *                 <p/>
+     *                 <p/>
+     *                 For example:
+     *                 http://bnc.lt/AUSTIN28.
+     *                 Should not exceed 128 characters
+     * @param type     An {@link int} that can be used for scenarios where you want the link to
+     *                 only deep link the first time.
+     * @param duration An {@link int} the time that Branch allows a click to remain outstanding
+     *                 and be eligible to be matched with a new app session.
+     * @param tags     An iterable {@link Collection} of {@link String} tags associated with a deep
+     *                 link.
+     * @param channel  A {@link String} denoting the channel that the link belongs to. Should not
+     *                 exceed 128 characters.
+     * @param feature  A {@link String} value identifying the feature that the link makes use of.
+     *                 Should not exceed 128 characters.
+     * @param stage    A {@link String} value identifying the stage in an application or user flow
+     *                 process. Should not exceed 128 characters.
+     * @param params   A {@link JSONObject} value containing the deep linked params associated with
+     *                 the link that will be passed into a new app session when clicked
+     * @param duration An {@link int} the time that Branch allows a click to remain outstanding
+     *                 and be eligible to be matched with a new app session.
+     * @param callback A {@link BranchLinkCreateListener} callback instance that will trigger
+     *                 actions defined therein upon receipt of a response to a create link request.
+     * @see BranchLinkData
+     * @see BranchLinkData#putTags(Collection)
+     * @see BranchLinkData#putChannel(String)
+     * @see BranchLinkData#putFeature(String)
+     * @see BranchLinkData#putStage(String)
+     * @see BranchLinkData#putParams(String)
+     * @see BranchLinkData#putDuration(int)
+     * @see BranchLinkCreateListener
+     * @deprecated use {@link io.branch.referral.builders.BranchShortUrlBuilder} instead.
+     */
+    @Deprecated
+    public String generateShortLink(final String alias, final int type, final int duration, final Collection<String> tags, final String channel, final String feature, final String stage, final String params, BranchLinkCreateListener callback, boolean async) {
+        ServerRequestCreateUrl req = new ServerRequestCreateUrl(context_, alias, type, duration, tags,
```

Comment:
Wait, the builder is internally calling a deprecated method? How is that going to work when you removed that method? You should extract that logic out to the builder and then update the deprecated methods within Branch to call the builder. That will allow you to easily remove these methods in the future, and make it so you don't need logic in more than one place.